components/servo/servo_delegate.rs

```diff
@@ -2,11 +2,29 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 use base::generic_channel;
-use embedder_traits::Notification;
+use embedder_traits::{Notification, RegisterOrUnregister};
+use url::Url;
 
 use crate::Servo;
 use crate::webview_delegate::{AllowOrDenyRequest, WebResourceLoad};
 
+pub struct ProtocolHandlerRequest {
+    pub scheme: String,
+    pub url: Url,
+    pub register_or_unregister: RegisterOrUnregister,
+    pub(crate) allow_deny_request: AllowOrDenyRequest,
+}
+
+impl ProtocolHandlerRequest {
+    pub fn allow(self) {
+        self.allow_deny_request.allow();
+    }
+
+    pub fn deny(self) {
+        self.allow_deny_request.deny();
+    }
+}
+
```

Comment:
With the way this is designed now the request will never be denied or allowed if the delegate method isn't implemented. Please add a `Drop` implementation here that denies the request and then just remove the implementation in servoshell.

Reply:
> We don't do this for other requests such as `PermissionRequest`. I guess it's missing there as well? If that's the case, we should probably ensure that `Drop` is implemented for all of these with a follow-up PR.

`PermissionRequest` contains an `AllowDenyRequest` which contains an `IpcResponder` which sends a default response on `Drop`. It's a bit tricky, but it should work properly.

Reply:
```
error[E0509]: cannot move out of type `webview_delegate::ProtocolHandlerRequest`, which implements the `Drop` trait
   --> components/servo/webview_delegate.rs:171:9
    |
171 |         self.allow_deny_request.deny();
    |         ^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         cannot move out of here
    |         move occurs because `self.allow_deny_request` has type `webview_delegate::AllowOrDenyRequest`, which does not implement the `Copy` trait
```
Will add a separate argument for the `AllowDenyRequest` to make this compile.

Reply:
You could also just make this an `Option` and take it when `deny()` or `allow()` is called.